src/library/pia/pia-utils.ts

```diff
@@ -0,0 +1,126 @@
+import * as PiaTypes from "./pia-types";
+import dayjs from "dayjs";
+
+export function piaSubstr(str: string, start: number, end: number): string {
+  let start_pos = start - 1;
+  let str_len = end - start + 1;
+  return str.substr(start_pos, str_len);
+}
+
+export function parsePiaString(
+  str: string,
+  start: number,
+  end: number
+): string {
+  return piaSubstr(str, start, end);
+}
+export function parsePiaMonthYear(
+  lineStr: string,
+  start: number,
+  end: number
+): PiaTypes.PiaMonthYear {
+  let moyrStr = piaSubstr(lineStr, start, end);
+  let month = moyrStr.slice(0, 2);
+  let year = moyrStr.slice(2);
+  // console.log(moyrStr, "month", month, "year", year);
+  /* starts in mmyyyy, dayjs takes yyyy-mm-dd */
+  var piaMonth: PiaTypes.PiaMonthYear = dayjs(
```

Comment:
Looks like most of these variables can all be defined with `const` rather than `let` since they don't change; ends up being more readable too when variables don't change. Definitely recommend dropping `var` though, since it's an old construct that doesn't respect block scoping, and can cause weird bugs down the line.

Reply:
I didn't realize about the block scoping change difference! Thanks for mentioning that.